### ISSUE TITLE
`preact-cli` isn't able to SSR if the app uses `@preact/signals`

### ISSUE THREAD

**Issue by rakeshpai** [preactjs/signals#168]
Just started to use signals in an app I created using preact-cli, but now the SSR breaks with the following error:

```
❯ npm run build

> my-project@0.0.0 build
> preact build

 Build  [==================  ] 92% (4.8s) chunk asset optimizationUnable to read file: /Users/rakesh.pai/code/my-project/src/src/index.ts

TypeError: Cannot read properties of null (reading '_u')
Cannot determine error position. This most likely means it originated in node_modules.

This could be caused by using DOM or Web APIs.
Pre-render runs in node and has no access to globals available in browsers.

Consider wrapping code producing error in: "if (typeof window !== "undefined") { ... }"

Alternatively use "preact build --no-prerender" to disable prerendering.

See https://github.com/preactjs/preact-cli#pre-rendering for further information.
```

This error message itself isn't very useful (and I don't know why there's an additional `/src` in the file path), but this error doesn't occur and the build runs successfully if I remove signals.

The usage of signals is very minimal:

```ts
import { computed, signal } from '@preact/signals';

const currentState = signal('idle');

const isFormEnabled = computed(() => currentState.value !== 'loading');

const SomeComponent = () => (
  <form>
    <input type="text" disabled={!isFormEnabled.value} />
  </form>
)
```

Am I doing something wrong?

**Comment by rakeshpai** [comment]
Also, please let me know if this error belongs in `preact-cli` instead, and I could raise it there.

**Comment by rschristian** [comment]
> This error message itself isn't very useful (and I don't know why there's an additional /src in the file path)

Sorry about that. Getting useful stack traces out of content built with Webpack has always been rather difficult, though I do spot a couple obvious issues and will try to spend some time on that.

The actual error position is in `../src/index.ts`. Our error message incorrectly assumes it's in your source code, and so prepends `/Users/rakesh.pai/code/my-project/src/`, hence the double `src`.

Not sure what the actual issue is, just speaking to the error at the moment.

**Comment by chrstntdd** [comment]
I'm running into this issue as well with a custom vite SSR setup.

Error gets thrown around here

```
	c("unmount", function (r, n) {
		var t = n.__c,
			i = t && v.get(t)
		if ((i && (v.delete(t), i._()(!0, !0)), "string" == typeof n.type)) {
			var e = n.__e,
				o = e._u  // <--- TypeError: Cannot read properties of null (reading '_u')
			o && (o._()(!0, !0), (e._u = null))
		}
		r(n)
	}),
```

This may also be rooted in `preact-render-to-string` too? I see some paths in the stack seem to indicate so.

```
node_modules/.pnpm/preact-render-to-string@5.2.4_preact@10.11.0/node_modules/preact-render-to-string/dist/commonjs.js:446:35)
```

**Comment by rakeshpai** [comment]
@rschristian 👍 There's no need to apologise, of course. I understand that this stuff is hard.

**Comment by rakeshpai** [comment]
@chrstntdd On an unrelated note, I've been trying to set up SSG with vite + preact, but have been hitting dead ends. I migrated to vite since I posted this issue, thinking it might be preact-cli's fault, but vite lacks SSG out of the box all together, which I'd really like. Any chance your code is open, and I could take a look? If not, could you tell me how I could go about it? Maybe this could be a part of `@preset/preact-vite` to be feature-compatible with `preact-cli`.

**Comment by rschristian** [comment]
> Maybe this could be a part of `@preset/preact-vite` to be feature-compatible with `preact-cli`.

And that's certainly something we're interested in having (at some point, finding time is hard). We'd like to offer up a migration path from `preact-cli` & `wmr` that's a bit closer to 1:1 with the feature sets.

**Comment by marvinhagemeister** [comment]
Fixed by #198
